### PR TITLE
Add API docs

### PR DESCRIPTION
**Pull request by null-a**
This PR adds some documentation for the public interface. (#4) There's lots of room to improve this, but I think it's ok for a first pass. What do you think?

### CODE REVIEW

File: brmp/fit.py [hunk 1]
```diff
@@ -177,6 +213,26 @@ def compute_diag_or_default(diag, samples):
...
+    distibutions of the parameters of the fitted model.
+
+    :param fit: A model fit.
+    :type fit: brmp.fit.Fit
+    :param qs: A list of quantiles to include in the output.
+    :type qs: list
+    :return: A table of marginal statistics.
+    :rtype: brmp.fit.ArrReprWrapper
+
+    Example::
```
Comment (neerajprad) [inline brmp/fit.py]: I just had a more general question - is there a reason why functions like `marginals` and `get_scalar_params` are not methods of the `Fit` class?

File: brmp/__init__.py [hunk 2]
```diff
@@ -54,16 +86,46 @@ def __init__(self, formula, metadata, contrasts, desc, data):
         self.metadata = metadata
         self.contrasts = contrasts
         self.desc = desc
+        # TODO: Turn this into a `@property` to improve generate docs?
```
[the remark this answers is not in the document]
Comment (null-a) [reply brmp/__init__.py]: The reason I omitted this was that I was trying to make something for end users, and the intention is that they never need to construct these directly. The core interface is pretty small, and if we start including stuff outside of it in the docs my guess it that the core will be quickly obscured, making the docs significantly less useful. There are probably better approaches to manage this though -- I'll pick this up over in #4.

File: brmp/formula.py [hunk 3]
```diff
@@ -74,10 +74,47 @@ def union(self, other):
 # using the name of the grouping column, which would be ambiguous
 # without the assumption. brms does this too.
 
+# TODO: I don't think attaching these docs to the "private" `Formula`
```
[the remark this answers is not in the document]
Comment (null-a) [reply brmp/formula.py]: Thanks, let's stick with this for now then.

> That's useful both for the users and other contributors who are working on these internal parts of the codebase.

I think there's a tension between the needs of these two groups, which is the root of my unease with what I'd done here. I've added a comment to #4 for further discussion.

File: brmp/__init__.py [hunk 4]
```diff
@@ -104,13 +166,61 @@ def _run_algo(self, algo, *args, **kwargs):
...
+
+        Example::
+
+          fit = defm('y ~ x', df).generate().prior()
+
+        """
         return self._run_algo('prior', num_samples, *args, **kwargs)
 
     def nuts(self, iter=10, warmup=None, num_chains=1, *args, **kwargs):
+        """
```
[the remark this answers is not in the document]
Comment (null-a) [reply brmp/__init__.py]: I'm not sure we have to choose between these two things? Among those users who know of both approaches, I guess that `.fit()` will be used most often. But I'm also ok with having `.nuts` etc. be considered part of the public interface. Do you foresee problems?

File: brmp/fit.py (hunk 1 above)
Comment (null-a) [reply brmp/fit.py]: No, there's nothing preventing us making this change. (And there's probably a bunch more non-idiomatic python waiting to be uncovered!)

**Pull request merged**